src/form/time/dropdown/TimeDropdown.tsx

```diff
@@ -0,0 +1,81 @@
+import "./_time-dropdown.scss";
+
+import React, {useEffect, useState} from "react";
+import classNames from "classnames";
+
+import Dropdown from "../../../dropdown/Dropdown";
+import {DropdownOption} from "../../../dropdown/list/item/DropdownListItem";
+import {generateTimeDropdownOptions} from "./util/timeDropdownUtils";
+
+export type TimeDropdownOption = DropdownOption<
+  string,
+  {date: Date; differenceInMinutes: number}
+>;
+
+export interface TimeDropdownProps {
+  testid: string;
+  selectedOption: null | TimeDropdownOption;
+  onSelect: (option: null | TimeDropdownOption) => void;
+  startDate?: Date;
+  startTimeString?: string;
+  customHeader?: React.ReactNode;
+  icon?: React.ReactNode;
+  hasDeselectOption?: boolean;
+  isDisabled?: boolean;
+  customClassName?: string;
+}
+
+function TimeDropdown({
```

Comment:
Do we really need this component, it seems it doesn't add much customization. We can still provide the necessary utilities to build this component and it'll be enough. (`generateTimeDropdownOptions`)

Reply:
Yes, we don't need `TimeDropdown` and `TimeSelect` components for now.